scipy/stats/_stats_py.py

```diff
@@ -1040,6 +1040,62 @@ def tsem(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     return sd / np.sqrt(am.count())
 
 
+def rms(a, *, axis=0):
+    r"""Return an array of the root-mean-square value of the passed array.
+
+    The root-mean-square is also called the "quadratic mean".
+
+    Parameters
+    ----------
+    a : array_like
+        n-dimensional array of which to find rms.
+    axis : int or None, optional
+        Axis along which to operate. Default is 0. If None, compute over
+        the whole array `a`.
+
+    Returns
+    -------
+    rms : ndarray
+        Array of rms values.
+
+    Notes
+    -----
+    Input `a` is converted to an `np.ndarray` before taking the rms.
+    To avoid the possibility of unexpected conversions, convert `a` to an
+    `np.ndarray` of the desired type before using `rms`.
+
+    For complex inputs, the absolute value is taken first:
+    ``sqrt(mean(abs(a)**2))``._[1]
+
+    References
+    ----------
+    .. [1] Smith, J. O. (2007). "Mathematics of the Discrete Fourier Transform
+       (DFT): With Audio Applications".
+       https://ccrma.stanford.edu/~jos/st/Other_Lp_Norms.html
+
+    Examples
+    --------
+    >>> a = np.array([[6, 8, 3, 0],
+    ...               [3, 2, 1, 7],
+    ...               [8, 1, 8, 4],
+    ...               [5, 3, 0, 5],
+    ...               [4, 7, 5, 9]])
+    >>> from scipy import stats
+    >>> stats.rms(a)
+    array([5.47722558, 5.03984127, 4.44971909, 5.84807661])
+
+    To get rms of whole array, specify ``axis=None``:
+
+    >>> stats.rms(a, axis=None)
+    5.229722745997153
+
+    """
+    if np.ndim(a) == 0:
+        axis = None
+
+    return np.sqrt(np.mean(np.abs(a) ** 2, axis=axis))
```

Comment:
```suggestion
    return np.sqrt((np.abs(a) ** 2).mean(axis=axis))
```

After the `np.abs` we are guaranteed to have a numpy object. Calling the mean directly is a bit faster.

![image](https://user-images.githubusercontent.com/883786/192094976-f279f8ae-dafc-4128-aaf1-b4cced23d82d.png)